src/libstd/collections/hash/map.rs

```diff
@@ -1518,6 +1528,17 @@ impl<'a, K, V> Entry<'a, K, V> {
     #[stable(feature = "rust1", since = "1.0.0")]
     /// Ensures a value is in the entry by inserting the default if empty, and returns
     /// a mutable reference to the value in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::HashMap;
+    ///
+    /// let mut map: HashMap<&str, usize> = HashMap::new();
+    /// map.entry("poneyland").or_insert(12);
+    ///
+    /// assert_eq!(map["poneyland"], 12);
```

Comment:
I don't agree with you. This way, it's very clear that the value has been correctly set to this key. I can add an `assert_eq!` to check the returned value of `or_insert` but I don't think removing the current one is a good idea.